npm/podController_test.go

```diff
@@ -0,0 +1,372 @@
+	f.podController.podListerSynced = alwaysReady
+
+	for _, pod := range f.podLister {
+		f.kubeInformer.Core().V1().Pods().Informer().GetIndexer().Add(pod)
+	}
+}
+
+func (f *podFixture) ipSetSave(ipsetConfigFile string) {
+	//  call /sbin/ipset save -file /var/log/ipset-test.conf
+	f.t.Logf("Start storing ipset to %s", ipsetConfigFile)
+	if err := f.ipsMgr.Save(ipsetConfigFile); err != nil {
+		f.t.Errorf("TestAddPod failed @ ipsMgr.Save")
+	}
+}
+func (f *podFixture) ipSetRestore(ipsetConfigFile string) {
+	//  call /sbin/ipset restore -file /var/log/ipset-test.conf
+	f.t.Logf("Start re-storing ipset to %s", ipsetConfigFile)
+	if err := f.ipsMgr.Restore(ipsetConfigFile); err != nil {
+		f.t.Errorf("TestAddPod failed @ ipsMgr.Restore")
+	}
+}
+
+func newNPMgr(t *testing.T) *NetworkPolicyManager {
+	npMgr := &NetworkPolicyManager{
+		NsMap:            make(map[string]*Namespace),
+		PodMap:           make(map[string]*NpmPod),
+		RawNpMap:         make(map[string]*networkingv1.NetworkPolicy),
+		ProcessedNpMap:   make(map[string]*networkingv1.NetworkPolicy),
+		TelemetryEnabled: false,
+	}
+
+	// (TODO:) should remove error return
+	// (Question) Is it necessary? Seems that it is necessary. Without this, it will have panic in PodController
+	allNs, _ := newNs(util.KubeAllNamespacesFlag)
+	npMgr.NsMap[util.KubeAllNamespacesFlag] = allNs
+	return npMgr
+}
+
+// 1. clean up below function
+func newPod(name, ns, rv string, labels map[string]string, status *corev1.PodStatus) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            name,
+			Namespace:       ns,
+			Labels:          labels,
+			ResourceVersion: rv,
+		},
+		Status: *status,
+	}
+}
+func newPodStatus(phase string, podIP string) *corev1.PodStatus {
+	return &corev1.PodStatus{
+		Phase: corev1.PodPhase(phase),
+		PodIP: podIP,
+	}
+}
+
+func createPod() *corev1.Pod {
+	labels := map[string]string{
+		"app": "test-pod",
+	}
+	status := newPodStatus("Running", "1.2.3.4")
+	podObj := newPod("test-pod", "test-namespace", "0", labels, status)
+	podObj.Spec = corev1.PodSpec{
+		Containers: []corev1.Container{
+			corev1.Container{
+				Ports: []corev1.ContainerPort{
+					corev1.ContainerPort{
+						Name:          "app:test-pod",
+						ContainerPort: 8080,
+					},
+				},
+			},
+		},
+	}
+	return podObj
+}
+
+func getKey(podObj *corev1.Pod, t *testing.T) string {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(podObj)
+	if err != nil {
+		t.Errorf("Unexpected error getting key for pod %s: %v", podObj.Name, err)
+		return ""
+	}
+	return key
+}
+
+func addPod(t *testing.T, f *podFixture, podObj *corev1.Pod) {
+	f.podLister = append(f.podLister, podObj)
+	f.kubeobjects = append(f.kubeobjects, podObj)
+
+	f.newPodController()
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	f.kubeInformer.Start(stopCh)
+
+	// simulate pod add event and add pod object to sharedInformer cache
+	f.podController.addPod(podObj)
+	if podObj.Spec.HostNetwork {
```

Comment:
Instead of doing checks on the hostnetpod and returning, can we check the workqueue length like below 
f.podController.workqueue.Len()

Reply:
Good idea. applied.